daemon/exec.go

```diff
@@ -246,5 +246,11 @@ func (d *Daemon) Exec(c *Container, execConfig *execConfig, pipes *execdriver.Pi
 	execConfig.ExitCode = exitStatus
 	execConfig.Running = false
 
+	// There is no use in keeping the execConfig structure alive is the job is running detached,
+	// because the CLI doesn't output the ExecID and there is no way to attach to it.
+	if !execConfig.OpenStdin && !execConfig.OpenStdout && !execConfig.OpenStderr {
```

Comment:
its too bad we can't get to the ExecConfig.Detach flag instead because not having a tty isn't necessarily the same thing as begin detached (I think).  Wonder if ContainerExecCreate() should copy it into "execConfig".